src/widget.h

```diff
@@ -24,13 +24,16 @@ enum class widget_var : int {
     stamina,        // Current stamina 0-10000, greater being fuller stamina reserves
     fatigue,        // Current fatigue, integer
     mana,           // Current available mana, integer
+    morale,         // Current morale level, integer (may be negative)
+    weariness,      // Current weariness, integer
```

Comment:
Can you call it weariness level? Weariness and wearine-- level are different concept- betisd the hood, and this is just tme level.

Reply:
Can you clarify the distinction between weariness and weariness level? I'm afraid I'm fuzzy on that.

The idea with all these is that the "var" name refers to a numeric attribute, which may be rendered as a literal decimal value (like focus, mana, or stats), or as a graph (like HP or stamina). My eventual plan was to allow a text rendering too (like `get_hunger_description` and `weariness_description` do) but haven't come up with a clean way to do it yet. Ideally I would prefer to call the var "weariness" regardless of which display style is used; the underlying value would remain a plain integer.

Reply:
So, there's weariness level, weariness, and weariness threshold.
Weariness level is a number somewhere on the scale of 0-5+, corresponding the activity levels it penalizes (e.g. no penalty is level 0, all activities penalized is level 5+, only extreme activities are penalized at level 1).
Weariness is a number that is used to generate weariness level by repeatedly subtracting the weariness threshold from it (and the threshold is multiplied by 0.8 each time) (weariness level is the number of times the threshold is subtracted), and derived from calories spent/gained, primarily.

Looking at our current display
![image](https://user-images.githubusercontent.com/42699974/130299287-92ecec53-87fd-4a81-8b3a-77ec14f399d5.png)
`weariness level` is `Weariness`
`weariness` is represented by the `Weary Transition` bar, kind of. It's actually the remainder of the weariness after the threshold can no longer be subtracted from it, and it shows you how far you have to go to either gain a level or lose one.

I haven't much looked at widgets (so I don't know what you need), but I think there are 5 important values regarding weariness:
1. `weariness level` - as described above 
2. `weariness` - as described above
3. `weariness to lvl 5` - the number that, if your `weariness` was that number, you would have a `weariness level` of 5 (all activities penalized)
4. `weariness remainder` as described above, the number used in the weary transition bar
5. `current weariness threshold` not the actual weariness threshold, but the weariness threshold multiplied by `0.8^weariness level` (something like that).

1 Can be used to generate `Weariness` in the current display (and is the return value of `Character::weariness_level` https://github.com/CleverRaven/Cataclysm-DDA/blob/ea1f5e19c87e94059e1e57e4f0fb2476129fa3c5/src/character.cpp#L6009
2 This is a new thing, with no existing equal, but is just the return value of `Character::weariness`
https://github.com/CleverRaven/Cataclysm-DDA/blob/ea1f5e19c87e94059e1e57e4f0fb2476129fa3c5/src/character.cpp#L5972
3 Same deal - I imagine this is used with 2 in a bar style display. You'd need to calculate this on your own. Also, it'd be messy because we don't yet do the thing where we cap weariness, so it has weird behavior when you get past level 5.
4 This is the first member returned from `Character::weariness_transition_progress`
5 This is the second member returned from `Character::weariness_transition_progress`
https://github.com/CleverRaven/Cataclysm-DDA/blob/ea1f5e19c87e94059e1e57e4f0fb2476129fa3c5/src/character.cpp#L5991-L6007

Hopefully that answers all your questions.

Reply:
Many thanks @anothersimulacrum for the comprehensive reply. I don't feel I will ever understand weariness but that brings me a few steps closer.

The main focus in the `widget` context is presenting information to the player; I chose the simpler name "weariness" to match the existing UI label, which are mapped from `weariness_level` to "Fresh", "Light", "Moderate" etc. in `panels.cpp`:

https://github.com/CleverRaven/Cataclysm-DDA/blob/ea1f5e19c87e94059e1e57e4f0fb2476129fa3c5/src/panels.cpp#L998

Side note, I would like to move this function into `Character` along with similar `get_thirst_description`, `get_weight_description` and so on - this would make writing a "weariness description" widget trivial. I expect to do that in a near-future PR.

But on reflection, and in anticipation of having a much larger selection of `widget_var`s in the future, I agree that `weariness_level` is the right choice here. I will make the adjustment.

Thanks again for taking the time to review and explain!

Reply:
I realize now I actually forgot some things which would be also be good pieces of information to have related to weariness, but aren't it's state - current activitiy level, max activity level (due to weariness), and the time penalty of the action you just took due to weariness (current activity level / max activity level).

Reply:
I hope to add `activity_level` pretty soon too.

I've also changed "morale" to "morale_level" and force-pushed.